### ISSUE TITLE
The 4 FoW buttons (right of A* toggle) don't do anything for players

### ISSUE THREAD

**Issue by Merudo** [RPTools/maptool#656]
**Describe the bug**
Clicking on the 4 FoW buttons (between A* toggle and "Select Map") don't do anything for players.

From my testing, the buttons only seem to work for the GM, and only when individualized views is enabled.

**Expected behavior**
The FoW buttons show/hide FoW depending on PC/NPC status, OR, the buttons just don't show up for players.

**MapTool Info**
- Version: 1.5.3, 1.5.4 RC1
- Install: New

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 10

**Additional context**
I recommend the FoW buttons be hidden to players, or they might accidentally click a button and lose vision.

**Comment by JamzTheMan** [comment]
It's a little miss-leading but they do work. They filter on ownership so a player could see what "he" could see with his PC token vs what his "horse" could see (NPC token).

But ya, its for individual views only and don't think the ME anything maybe for players?  We def could filter/hide these based on server settings.

**Comment by Phergus** [comment]
So it sounds like we should disable them on clients if the server is not set for Individual Views.

**Comment by JamzTheMan** [comment]
ya, sounds reasonable.

**Comment by Phergus** [comment]
Okay then.  Code monkey volunteers have at it.

**Comment by Merudo** [comment]
Thanks for the explanation! It clears things up.

Note that there might be an issue for the ME button for players - it seems it shows the same FoW as the ALL button.

**Comment by emmebi** [comment]
Code monkey here! Since this has been recently mentioned/asked in the forum, can you assign it to me? Unless someone else is already working on it.

**Comment by Phergus** [comment]
It's all yours!

**Comment by JamzTheMan** [comment]
The ME button was original the GM button. For the GM, it show fow only for tokens that have `no ownership`, eg, doesn't show PC vision.

For players, it probably could be hidden? I'd double check the code but probably right that ME is same as ALL for players (maybe ALL should be removed for players instead, probably makes more sense?)

**Comment by JamzTheMan** [comment]
Thinking about it more... the `ALL` button may actually show all tokens that a player owns OR is `owned by all` where as the ME button may only show tokens specifically owned by a player name.

**Comment by JamzTheMan** [comment]
Bottom line, there was either a method to my madness or a madness to my method.......

**Comment by JamzTheMan** [comment]
Ok tested and there was a method to my madness!

`ME` button only shows FoW for tokens I specifically own, eg my PC token owned by `Jamz`. 
`ALL` also shows me vision of tokens `Owned by All` e.g. that NPC villager the GM gave us ownership to all the players so anyone can move around and we have to keep alive. Furthermore you have a PC token owned by all that is, i don't know, some magic orb eyeball that floats around.

To recap:

ME: Show tokens owned by Jamz
All: Show tokens owned by Jamz OR Owned by All
PC: Show tokens owned by Jamz OR Owned by All that are PC tokens
NPC: Show tokens owned by Jamz OR Owned by All that are NPC tokens

**Comment by Merudo** [comment]
Yes, it does appear to work.

I don't know if the buttons got fixed, or if I did my initial testing on 1.5.3 incorrectly.

**Comment by JamzTheMan** [comment]
The buttons always worked that way but there was some funkyness with iView/iFoW and we added that gm reveals options and what not.

It's still valid to disable/remove the buttons though when they actually do NOT do anything, eg. NOT using iView.  Would need to test with the various server options matrix.

**Comment by Merudo** [comment]
Now that it has been established that the buttons work, I'm fine with keeping them.

**Comment by JamzTheMan** [comment]
Well, the work with the right server settings otherwise they just sit there :)

Would it be better to default it to ME instead? I was just going to suggest tooltips but duh, they are there and I didn't even use them lol  Maybe a wiki doc then?

**Comment by melek** [comment]
I've updated the Introduction to Lights and Sights with a section on Individual Views, including info about these buttons: https://lmwcs.com/rptools/wiki/Introduction_to_Lights_and_Sights#Individual_Views_Server_Setting

The information is linked and button functions duplicated on the User Interface article - I should rename this section to 'Individual View Vision Settings' since that is much more accurate, but I don't want to recreate any graphics at the moment so 'Fog of War Vision Settings' it remains:
https://lmwcs.com/rptools/wiki/MapTool_User_Interface#Fog_of_War_Vision_Settings

I also smoothed over the 'Individual Views' description in the Game Hosting article:
https://lmwcs.com/rptools/wiki/Introduction_to_Game_Hosting#Ownership.2C_Vision.2C_and_Behavior

For what its worth, I softly think it should continue to default to 'All', but I only say that because it is current default and it makes it easy to ignore these settings for most players.

**Comment by emmebi** [comment]
@JamzTheMan @Merudo , I came back to this after a while, and I am a little confused: do we need to disable some buttons for the player view, or only for specific conditions? And what are them?

**Comment by melek** [comment]
I believe the correct change is to hide the buttons for all clients when a server is running without 'Individual Views'. The PC/NPC buttons are functional on a personal server, so I guess leave them when not hosting - though 'Me' vs. 'All' seems currently interchangeable to the GM (probably since the GM is considered an 'owner' for all tokens). 

* Here is how the buttons function for a player when both Individual Views and Individual Fog of War is on: https://d.pr/v/CvNy0c
* Here is how the buttons (do not) function for a player when Individual Views is unchecked: https://d.pr/v/ZEjLDv
    Notice also that the 'NPC' tokens do not appear in this screencast, even if they are player-owned - just an interesting detail of how MapTool handles NPCs when individual views are off. This sometimes shows up in Discord as a support question, and we have to tell the user to switch player tokens' type to PC or their players won't be able to see it on a map with vision unless it is in sight range of a PC token.

Here is a screenshot of this sample map from the GM side with red to highlight the VBL boundary running a server with Individual Views off (I've also attached the map in case you want to use it for testing - useful with Token Labels on). Each token has a `circle 5` light-source and vision range, Vision set to 'Night':
![image](https://user-images.githubusercontent.com/220149/112728765-0e642b00-8ee6-11eb-99fc-e9da03fcc620.png)
Also interesting is that when no tokens are selected in the GM view, only PC lights/vision are shown.

I tried to figure out how to hide these properly (and failed) a few months ago, but couldn't figure it out. I know that `ToolbarPanel.java` is where the relevant code resides, but I tried to refactor the buttons so it would be easier to control their visibility and just couldn't get it to work - and definitely didn't know how to get it to only display conditionally based on server status.

[View Ownership Test Map.zip](https://github.com/RPTools/maptool/files/6216083/View.Ownership.Test.Map.zip)

**Comment by emmebi** [comment]
@melek when you mention All Client do you mean All **Player** Client ?

**Comment by melek** [comment]
I mean all clients, the buttons don't do anything for the GM either on a server with individual views off from my tests.

**Comment by emmebi** [comment]
After more time that I initially foresee, I actually have a PR for this; I went down the road of disabling the buttons rather than hiding them, because hiding them as the result seems better.

Can someone re-assign this to me so that I can publish the PR?
